internal/gps/source_manager.go

```diff
@@ -148,7 +148,7 @@ var _ SourceManager = &SourceMgr{}
 // gps's SourceManager is intended to be threadsafe (if it's not, please file a
 // bug!). It should be safe to reuse across concurrent solving runs, even on
 // unrelated projects.
-func NewSourceManager(cachedir string) (*SourceMgr, error) {
+func NewSourceManager(cachedir string, log func(v ...interface{})) (*SourceMgr, error) {
```

Comment:
The long term vision for the `NewSourceManager()` API is murky, which is part of why i want to avoid changing it impulsively now. but, yes, i imagine we're gonna head in a struct-like direction.

an unexported constructor that sets a logger for test purposes would be 👍 for now.